contracts/eosio.system/include/eosio.system/eosio.system.hpp

```diff
@@ -968,6 +981,40 @@ namespace eosiosystem {
          [[eosio::action]]
          void setacctcpu( const name& account, const std::optional<int64_t>& cpu_weight );
 
+         /**
+          * Computes a hash for a vector of names
+          *
+          * @param patterns - vector of names.
+          */
+         [[eosio::action]]
+         checksum256 denyhashcalc( const std::vector<name>& patterns );
```

Comment:
Please note in the comments that this is a read-only action.

Reply:
https://github.com/eosnetworkfoundation/eos-system-contracts/pull/191/commits/587f31877ae1280f9ca25967969597d429caf10a